From petkan@nucleusys.com Wed Mar 29 22:59:00 2006
Date: Thu, 30 Mar 2006 09:59:22 +0300 (EEST)
From: Petko Manolov <petkan@nucleusys.com>
To: Greg KH <greg@kroah.com>
Subject: USB: pegasus driver bugfix
Message-ID: <Pine.LNX.4.62.0603300953150.7186@localhost.localdomain>

Attached is a patch that fixes nasty bug, which i am afraid was there
for a long time.  It was spotted by Andre Draszik <kernel@andred.net>.

From: Petko Manolov <petkan@nucleusys.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/net/pegasus.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- gregkh-2.6.orig/drivers/usb/net/pegasus.c
+++ gregkh-2.6/drivers/usb/net/pegasus.c
@@ -262,7 +262,7 @@ static int set_register(pegasus_t * pega
 	usb_fill_control_urb(pegasus->ctrl_urb, pegasus->usb,
 			     usb_sndctrlpipe(pegasus->usb, 0),
 			     (char *) &pegasus->dr,
-			     &tmp, 1, ctrl_callback, pegasus);
+			     tmp, 1, ctrl_callback, pegasus);
 
 	add_wait_queue(&pegasus->ctrl_wait, &wait);
 	set_current_state(TASK_UNINTERRUPTIBLE);
